### ISSUE TITLE
Add support for Postgres sequences

### ISSUE THREAD

**Issue by RiderSx** [analogueorm/analogue#44]
Okay, let me explain why is this really MUST be implemented.

As you may know Postgres uses sequences to generate auto-incremented id's, that's more reliable method to implement auto-increments than MySQL's one. This gives us more control over lastInsertedId command.

Why you really need to implement this feature? Because of the way how Postgres executes commands, sometimes there is something else saving in paraller of other row and that gives a collision (from our perspective), more known as race condition, in that situation when we call lastInsertedId w/o defining from which sequence we want it we will get id of a wrong row in wrong table or get just NULL because row saved in parallel has no primary key.

So in conclusion: sequence feature will eliminate race conditions.

Here is a quote from a [php's manual for PDO::lastInsertId](http://pt2.php.net/manual/en/pdo.lastinsertid.php):
> Returns the ID of the last inserted row, or the last value from a sequence object, depending on the underlying driver. For example, PDO_PGSQL requires you to specify the name of a sequence object for the name parameter.

[Laravel already supports it](https://github.com/laravel/framework/blob/5.0/src/Illuminate/Database/Query/Builder.php#L1719)

**Comment by RemiCollin** [comment]
I wasn't aware of this Postgres behaviour. 

Which value should be provided to the sequence parameter ? Some UID ?

**Comment by RiderSx** [comment]
It should be configurable like table name. By default Postgres makes names %table_name%_id_seq.

**Comment by RemiCollin** [comment]
Is there a need to use the CREATE SEQUENCE statement before it ? Or just the sequence name will do ?

**Comment by RiderSx** [comment]
Just sequence name, no need to create it because it is already created when query time comes.

**Comment by RemiCollin** [comment]
Ok, I'll add this in the next batch. Thanks for pointing this out :)

**Comment by RiderSx** [comment]
Thanks for fast response, appreciate it :)

**Comment by RemiCollin** [comment]
Added in https://github.com/analogueorm/analogue/pull/54

**Comment by RiderSx** [comment]
:+1:
